util/topgen/merge.py

```diff
@@ -100,14 +142,22 @@ def amend_ip(top, ip):
             # Removing descriptors, checking for security-relevant parameters
             # that are not exposed, adding a top-level name.
             for i in ip_module["param_list"]:
-                i.pop("desc", None)
                 par_name = i["name"]
                 if par_name.lower().startswith("sec"):
                     log.warning("{} has security-critical parameter {} "
                                 "not exposed to top".format(mod_name, par_name))
```

Comment:
@vogelpi quick question: this seems to do the exact opposite of what the message says... i.e., it lists all `Sec` parameters that are exposed at the top... should we fix that (i.e., by removing "not" in the message)?

Reply:
Uh, that's actually a bug. The expected behavior is that topgen prints the warning whenever it finds a `Sec` parameter that is NOT exposed. So the if condition should be extended with something like `and i["expose"] == False` .

At the moment, I think it prints a warning about every `Sec` parameter, right?

Reply:
yes it prints out all sec parameters

Reply:
I'll fix that

Reply:
ok this is now fixed.